packages/block-editor/src/store/reducer.js

```diff
@@ -822,7 +822,9 @@ export const blocks = flow(
 					( accumulator, id ) => ( {
 						...accumulator,
 						[ id ]: reduce(
-							action.attributes,
+							action.attributes[ id ]
```

Comment:
It seems a bit weird to rely on the "id" for this check. What if there's an attribute  that conflicts with the  "id" (both are strings). I wonder if it's better to be explicit here and have a separate action (or a flag).

cc @ellatrix

Reply:
@youknowriad I guess I was thinking that there was no chance of an attribute name being in the format of `"b4c1d31e-fba3-4438-8d87-8d3554d8c4fb"` and randomly matching an actual block id.

I think you are right though, the API is better if we are more explicit. It seems like the two options are too close to warrant a whole separate action so have added a `uniqueByBlock` flag. Let me know if you think this is a suitable way forward and I will extend the unit tests.